libraries/AP_Common/AP_ExpandingArray.h

```diff
@@ -95,15 +95,23 @@ class AP_ExpandingArray : public AP_ExpandingArrayGeneric
     T &operator[](uint16_t i)
     {
         const uint16_t chunk_num = i / chunk_size;
-        const uint16_t chunk_index = (i % chunk_size) * elem_size;
-        return (T &)(chunk_ptrs[chunk_num][chunk_index]);
+        const uint16_t chunk_index = (i % chunk_size);
+        #pragma GCC diagnostic push
```

Comment:
nitpick: slightly indenting issue with the top pragma

Reply:
fixed